tests/typed_agent.py

```diff
@@ -178,6 +178,13 @@ def run_sync3() -> None:
     assert_type(result.data, Union[Foo, Bar])
 
 
+MyUnion: TypeAlias = 'Foo | Bar'
+union_agent2: Agent[None, MyUnion] = Agent(
+    result_type=MyUnion,  # type: ignore[arg-type]
```

Comment:
Why do we need the type ignore here? What's the error?

Reply:
we need the type here so we can enforce the return type. The error is that a union is not a type, but a typeform, docs about waiting for PEP 747 - https://ai.pydantic.dev/results/?h=pep#structured-result-validation